packages/page-bounties/src/BountyInitiateVoting.tsx

```diff
@@ -0,0 +1,107 @@
+// Copyright 2017-2020 @polkadot/app-treasury authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import type { SubmittableExtrinsic } from '@polkadot/api/types';
+import type { DeriveCollectiveProposal } from '@polkadot/api-derive/types';
+
+import BN from 'bn.js';
+import React, { useEffect, useMemo, useState } from 'react';
+
+import { getTreasuryProposalThreshold } from '@polkadot/apps-config';
+import { Button, InputAddress, Modal, TxButton } from '@polkadot/react-components';
+import { useApi, useMembers, useToggle } from '@polkadot/react-hooks';
+
+import { useTranslation } from './translate';
+
+interface Props {
+  index: number;
+  proposals?: DeriveCollectiveProposal[];
+}
+
+const BOUNTY_METHODS = ['approveBounty', 'closeBounty'];
+
+function BountyInitiateVoting ({ index, proposals }: Props): React.ReactElement<Props> | null {
+  const { t } = useTranslation();
+  const { api } = useApi();
+  const { isMember, members } = useMembers();
+  const [isOpen, toggleOpen] = useToggle();
+  const [accountId, setAccountId] = useState<string | null>(null);
+  const [threshold, setThreshold] = useState<BN>();
+
+  useEffect((): void => {
+    members && setThreshold(
+      new BN(Math.ceil(members.length * getTreasuryProposalThreshold(api)))
+    );
+  }, [api, members]);
+
+  const approveBountyProposal: SubmittableExtrinsic<'promise'> = (api.tx.bounties || api.tx.treasury).approveBounty(index);
+  const closeBountyProposal: SubmittableExtrinsic<'promise'> = (api.tx.bounties || api.tx.treasury).closeBounty(index);
```

Comment:
Don't believe you want to create this each render? (They are quite heavy, a new class instance). Rather create the full proposal - probably in the effect above. then just pass it as `extrinsic={approve/close}` instead of `tx/params`

Reply:
Would the useRef do the work?